packages/pwa-kit-react-sdk/src/ssr/universal/hooks/use-get-props-args.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useContext} from 'react'
+import {useParams, useLocation} from 'react-router-dom'
+import AppConfig from '../../universal/components/_app-config'
+
+export interface GetPropsArgs {
+    location: Location
+    params: {[key: string]: string}
+    req: object | undefined // Get type from runtime?
+    res: object | undefined // Get type from runtime?
```

Comment:
```suggestion
    req?: object // Get type from runtime?
    res?: object // Get type from runtime?
```
`req: object | undefined` means "`req` _must_ be defined, but might be explicitly set to `undefined` (i.e. `{req: undefined}`).
`req?: object` means "`req` may or may actually be there" (i.e. `{}`).